fdbserver/storageserver.actor.cpp

```diff
@@ -14718,17 +14653,23 @@ ACTOR Future<Void> storageEngineConsistencyCheck(StorageServer* self) {
 		}
 
 		CoalescedKeyRangeMap<std::string> currentShards;
+		std::unordered_map<std::string, int> teamShardCount;
 		currentShards.insert(allKeys, "");
 		auto fullRange = self->shards.ranges();
 		for (auto it = fullRange.begin(); it != fullRange.end(); ++it) {
 			if (!it.value()) {
 				continue;
 			}
 			if (it.value()->assigned()) {
-				currentShards.insert(it.range(), format("%016llx", it.value()->shardId));
+				currentShards.insert(it.range(), format("%016llx", it.value()->getShardId()));
+				teamShardCount[it.value()->getTeamId()]++;
 			}
 		}
 
+		for (const auto& kv : teamShardCount) {
+			TraceEvent("StorageServerTeamShardCount").detail("TeamId", kv.first).detail("ShardCount", kv.second);
```

Comment:
How many events can be generated? If a lot, this might be throttled.

Reply:
> How many events can be generated? If a lot, this might be throttled.

Expect no more than 40 events each run. It's rare to have storage servers with more than 20 teams. Team remover will be triggered in that case.